src/components/TodosLogic.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import TodoInput from './TodoInput';
+import TodosList from './TodosList';
+
+const getInitialTodos = () => {
+  const temp = localStorage.getItem('todos');
+  const loadedTodos = JSON.parse(temp);
+  return loadedTodos || [];
+};
+
+const TodosLogic = () => {
+  const [todos, setTodos] = useState(getInitialTodos());
+
+  useEffect(() => {
+    const temp = JSON.stringify(todos);
+    localStorage.setItem('todos', temp);
+  }, [todos]);
+
+  const handleTodoItem = (id) => {
+    setTodos((prevTodos) => prevTodos.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+      return todo;
+    }));
+  };
+
+  const handleRemoveTodo = (id) => {
+    setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== id));
+  };
+
+  const addTodo = (title) => {
+    const newTodo = {
+      id: todos.length + 1,
+      title,
+      completed: false,
+    };
+    setTodos([...todos, newTodo]);
+  };
+
+  const editTodo = (updatedTitle, id) => {
+    const updatedTodos = todos.map((todo) => {
+      if (todo.id === id) {
+        const updateTitle = updatedTitle;
+        return { ...todo, title: updateTitle };
+      }
+      return todo;
+    });
+    setTodos(updatedTodos);
+  };
```

Comment:
Optional:  
 ♻️ Maybe you can improve the readability of this part using something like:

const editTodo = (updatedTitle, id) => {
  setTodos(todos.map(todo => (
    todo.id === id ? { ...todo, title: updatedTitle } : todo
  )));
};

Reply:
Thanks @mrcbq I'll take it into consideration.